pkg/template/apis/template/v1/types.go

```diff
@@ -167,8 +170,19 @@ const (
 	// TemplateInstanceInstantiateFailure indicates the failure of the template
 	// instantiation
 	TemplateInstanceInstantiateFailure TemplateInstanceConditionType = "InstantiateFailure"
+	// TemplateInstanceWaiting indicates waiting for readiness of the template
+	// instantiation.
+	TemplateInstanceWaiting TemplateInstanceConditionType = "Waiting"
```

Comment:
Hrm, @kargakis should this be consistent with `Available` or other conditions that we have already defined?